python/python/tests/utils/utils.py

```diff
@@ -112,4 +128,21 @@ def compare_maps(
         return True
     if map1 is None or map2 is None:
         return False
-    return json.dumps(map1) == json.dumps(map2)
+    return json.dumps(convert_bytes_to_string_dict(map1)) == json.dumps(
+        convert_bytes_to_string_dict(map2)
+    )
+
+
+def convert_str_to_bytes_list(lst):
+    return [
+        (
+            elem.encode("utf-8")
+            if isinstance(elem, str)
+            else convert_str_to_bytes_list(elem) if isinstance(elem, list) else elem
+        )
+        for elem in lst
+    ]
+
+
+def convert_str_to_bytes_set(set):
```

Comment:
Are those similar to `convert_bytes_to_string_dict` but in opposite direction? Create a single function which does the same approach.